samtranslator/translator/translator.py

```diff
@@ -269,6 +274,83 @@ def _get_resources_to_iterate(
         return functions + statemachines + apis + others + connectors
 
 
+def add_embedded_connectors(sam_template: Dict[str, Any], document_errors: List[Any]) -> None:
+    """
+    Loops through the SAM Template resources to find any connectors that have been attached to the resources.
+    Adds those attached connectors as Connector resources to the list of resources
+
+    :param dict sam_template: SAM template
+    """
+    resources = sam_template.get("Resources", {})
+    connectors = {}
+
+    # Loop through the resources in the template and see if any connectors have been attached
+    for source_logical_id, resource in resources.items():
+        if "Connectors" in resource:
+            try:
+                sam_expect(
+                    resource["Connectors"], source_logical_id, "Connectors", is_resource_attribute=True
+                ).to_be_a_map()
+
+                # Go through each of the connectors that have been attached and create a Serverless Connector resource
+                for connector_logical_id, connector_dict in resource["Connectors"].items():
+
+                    # The logical id for the transformed connector will be <source_logical_id> + <embedded_connector_logical_id>
+                    connector_logical_id = source_logical_id + connector_logical_id
+
+                    # can't use sam_expect since this is neither a property nor a resource attribute
+                    if not isinstance(connector_dict, dict):
+                        raise InvalidResourceException(connector_logical_id, f"{connector_logical_id} should be a map.")
+
+                    # mutates the connector_dict to make it a connector resource
+                    get_generated_connector(source_logical_id, connector_logical_id, connector_dict, document_errors)
+
+                    connectors[connector_logical_id] = connector_dict
+
+                # delete connectors key, so it doesn't show up in the transformed template
+                del resource["Connectors"]
+            except InvalidResourceException as e:
+                document_errors.append(e)
+
+    # go through the list of generated connectors and add it to resources dict
+    for connector_logical_id, connector_dict in connectors.items():
+        if connector_logical_id in resources:
+            document_errors.append(
+                DuplicateLogicalIdException(
+                    connector_logical_id, connector_logical_id, resources[connector_logical_id]["Type"]
+                )
+            )
+        else:
+            resources[connector_logical_id] = connector_dict
+
+
+def get_generated_connector(
+    source_logical_id: str, connector_logical_id: str, connector: Dict[str, Any], document_errors: List[Any]
+) -> None:
```

Comment:
IMO would be clearer if this took the connector from the resource attribute as input, and returned the connector resource as output. Currently it mutates an input variable, making it hard to audit.